src/app/components/client/SubscriptionCheck.tsx

```diff
@@ -10,12 +10,11 @@ import { useEffect } from "react";
 export const SubscriptionCheck = () => {
   const { update } = useSession();
 
+  // Poll for the session every minute.
   useEffect(() => {
-    void update();
-
-    // This should only run once per page load - `update` will always appear to be changed.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    const interval = setInterval(() => void update(), 1000 * 60);
+    return () => clearInterval(interval);
```

Comment:
Unfortunately `setInterval` is notoriously tricky to align with React's rendering cycle. See https://overreacted.io/making-setinterval-declarative-with-react-hooks/

(There's a "Just Show Me The Code" section with a proper implementation.)

Also... Oof, it feels like somehow this shouldn't be necessary :( )